packages/utils/src/typedObjectKeys.ts

```diff
@@ -0,0 +1,2 @@
+export const typedObjectKeys = <T extends Record<any, any>>(obj: T): Array<keyof T> =>
```

Comment:
There's a catch to it
```
A common ask is to provide 'better' typings for Object.keys, so that it returns Array<keyof T> instead of Array<string>. Same for Object.entries. ts-reset won't be including rules to change this.

TypeScript is a structural typing system. One of the effects of this is that TypeScript can't always guarantee that your object types don't contain excess properties:

type Func = () => {
  id: string
}

const func: Func = () => {
  return {
    id: '123',
    // No error on an excess property!
    name: 'Hello!',
  }
}
So, the only reasonable type for Object.keys to return is Array<string>.
```

But I guess we can risk it. Since we always basically typecast it.

Reply:
Yeah, it's not correct. But it's not more broken than it already was either. :innocent: